src/main/java/com/example/decafe/Customer.java

```diff
@@ -239,45 +180,32 @@ public void run() {
 
     //Methode for the general 60 seconds timer
     public void waitingTime() {
-        Customer customer = this;
         TimerTask timerTask = new TimerTask() {
             int seconds = 60;
 
             @Override
             public void run() {
-                seconds--;
-                if (seconds == 59) { //set green smiley when the customer has just spawned
-                    smiley.setVisible(true);
-                    try {
-                        smiley.setImage(createImage("smileygreen.png"));
-                    } catch (FileNotFoundException e) {
-                        e.printStackTrace();
-                    }
-                    customer.setMood(CustomerMode.GREEN);
-                } else if (seconds == 30) { //set yellow smiley when the customer has just spawned
-                    smiley.setVisible(true);
-                    try {
-                        smiley.setImage(createImage("smileyyellow.png"));
-                    } catch (FileNotFoundException e) {
-                        e.printStackTrace();
-                    }
-                    customer.setMood(CustomerMode.YELLOW);
-                } else if (seconds == 15) { //set red smiley when the customer has just spawned
-                    smiley.setVisible(true);
-                    try {
-                        smiley.setImage(createImage("smileyred.png"));
-                    } catch (FileNotFoundException e) {
-                        e.printStackTrace();
-                    }
-                    customer.setMood(CustomerMode.RED);
-                } else if (seconds == 0) { //when the timer has finished - customer leaves
-                    startTimerLeave(customer);
+                seconds--; // increment after or before ?
+                if (seconds == 0) {
+                    startTimerLeave(Customer.this);
+                } else if (seconds == 59 || seconds == 30 || seconds == 15) {
+                    updateSmiley(seconds);
                 }
+
             }
         };
+        sixtySecondsTimer.schedule(timerTask, 0, 1000);
+    }
 
-        this.sixtySecondsTimer.schedule(timerTask, 0, 1000);//it should call this methode every second
-
+    private void updateSmiley(int seconds) {
+        try {
```

Comment:
Add the used image names to ImageAssets and call them from the ImageAssets enum : ImageAssets.SMILEY_GREEN.getName()